src/js/tech/tech.js

```diff
@@ -155,6 +155,30 @@ class Tech extends Component {
     }
   }
 
+  /**
+   * A special function to trigger source set in a way that will allow player
+   * to re-trigger if the player or tech are not ready yet.
+   *
+   * @fires Tech#sourceset
+   */
+  triggerSourceset() {
+    if (!this.isReady_) {
+      // on initial ready we have to trigger source set
+      // 1ms after ready so that player can watch for it.
+      this.one('ready', () => this.setTimeout(this.triggerSourceset, 1));
```

Comment:
to verify: if we have multiple source changes when we're not ready, when we get `sourceset`, would `player.currentSrc()` return the last source or each source?
Things that would affect it: how setting the source works when we're not ready. A potential workaround would be to add the source to the event object.

Reply:
hmm I think adding the source to the event would make sense instead of waiting for `ready`. We would just have to make sure that there is a way to do this:
```
var player = videojs(...);
player.on('sourceset', (e) => console.log('the first source was set');
```

I think that the source may be set before a listener can be added

Reply:
I can confirm that if there are two source changes when the tech isn't ready, and we check the source, we get the latest version, currently.
```js
player.on('sourceset', () => console.log(player.currentSource()));
player.src({src: './oceans.webm', type: 'video/webm'});
player.src({src: './oceans.mp4', type: 'video/mp4'});
```
we'd get two logs of the oceans.mp4.

Reply:
I think the only real solution is to add the source to the event and then document that if you want to know for sure what source was set when the event was triggered, you need to look at the event object, to know the latest one, look on the player, sort like how `currentTime` exists on the event object but no one uses it. We can also mention that this should only really be a problem for multiple sets around tech switching.

Reply:
Since removing this waiting and the timeout means that a user cannot do something like:
```js
player.ready(() => player.on('sourceset', () =>  console.log(player.currentSrc())))
```
or
```js
```js
player.on('sourceset', () => player.on('sourceset', () =>  console.log(player.currentSrc())))
```

Reply:
This is a really fun one. Turns out you'll end up getting `oceans.webm` after getting the `oceans.mp4` source because when `oceans.webm` is being triggered, the tech isn't ready yet but by the time that `oceans.mp4` is set, the tech is ready and so it ends up triggering immediately and then `oceans.webm` gets triggered shortly after.
So, you'll see something like:
```js
var player = videojs('vid', {
  techOrder: ['flash', 'html5']
});
// so that we'll change techs when we set a new source
player.src({ src: './oceans.flv', type: 'video/flv'});

player.on('sourceset', (e) => console.log(e.src, player.currentSrc()));
player.src({src: './oceans.webm', type: 'video/webm'});
player.src({src: './oceans.mp4', type: 'video/mp4'});
//> oceans.mp4, oceans.mp4
//> oceans.webm, oceans.mp4
```